scopesim/effects/spectral_trace_list_utils.py

```diff
@@ -457,12 +457,17 @@ def __init__(self, fov, dlam_per_pix):
         # Initialise the array to hold the xi-lambda image
         self.image = np.zeros((n_xi, n_lam), dtype=np.float32)
         self.lam = cube_lam
+        try:
+            dlam_per_pix_val = dlam_per_pix(np.asarray(self.lam))
+        except TypeError:
```

Comment:
When would this exception occur? `XiLamImage` is only initialized once in ScopeSim, earlier in this file. There `dlam_per_pix` is always set to the output of `interp1d`, which is always a function I think. So it seems this `except` block can be removed, which would clarify better what's going (perhaps with type hints on the parameters).

Reply:
This was meant to preserve the previous behaviour where `dlam_per_pix` was passed as just one number (the average dispersion).  Although in the current implementation this should not occur, I wouldn't exclude that `XiLamImage` could be used in a different context in the future.